### PR TITLE
Fix crash with UniqInjectiveFunctionsEliminationPass and uniqCombined

### LINKED ISSUE #65052: heap-buffer-overflow after uniqCombined
https://s3.amazonaws.com/clickhouse-test-reports/0/686d6d676ac13a2265538cce17db94f4602c749a/ast_fuzzer__asan_.html

```
SELECT sum(u)
FROM
(
    SELECT
        intDiv(number, 4096) AS k,
        uniqCombined(tuple(materialize(toLowCardinality(toNullable(16))))) AS u
    FROM numbers(4096 * 100)
    GROUP BY k
)
```

Does not happen if you replace uniqCombined by a different uniq* aggregate function

### PR DESCRIPTION
**Pull request by Algunenano**
### Changelog category (leave one):
- Critical Bug Fix (crash, LOGICAL_ERROR, data loss, RBAC)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
- Fix crash with UniqInjectiveFunctionsEliminationPass and uniqCombined

### Documentation entry for user-facing changes

Closes https://github.com/ClickHouse/ClickHouse/issues/65052

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: Normal Builds
- [ ]  Allow: Special Builds
- [ ]  Allow: All NOT Required Checks
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Do not test
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll3** [comment]
*This is an automated comment for commit b7161b77d177680187b489b971f35ab856a4004b with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/65188/b7161b77d177680187b489b971f35ab856a4004b/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>A Sync</td><td>If it fails, ask a maintainer for help</td><td>⏳ pending</td></tr>
<tbody>
</table>
<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>Checks correctness of the PR's body</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

**Review by devcrafter — APPROVED** [review]

File: src/Analyzer/Passes/UniqInjectiveFunctionsEliminationPass.cpp [hunk 1]
```diff
@@ -41,9 +41,9 @@ class UniqInjectiveFunctionsEliminationVisitor : public InDepthQueryTreeVisitorW
             return;
 
         bool replaced_argument = false;
-        auto & uniq_function_arguments_nodes = function_node->getArguments().getNodes();
+        auto replaced_uniq_function_arguments_nodes = function_node->getArguments().getNodes();
```
Comment (novikd) [inline src/Analyzer/Passes/UniqInjectiveFunctionsEliminationPass.cpp:44]: Why do you need to make a copy here?

Comment (Algunenano) [reply src/Analyzer/Passes/UniqInjectiveFunctionsEliminationPass.cpp:44]: In case we end up discarding the optimization because of the return type change.

**Comment by canhld94** [comment]
Should it be close #65052?

**Comment by Algunenano** [comment]
> Should it be close https://github.com/ClickHouse/ClickHouse/issues/65052?

Yes, I mistook the number and was closing the PR by mistake. Thanks!

**Pull request merged**